src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NativeLayoutVertexNode.cs

```diff
@@ -380,8 +380,9 @@ public static NativeLayoutTypeSignatureVertexNode NewTypeSignatureVertexNode(Nod
                     return new NativeLayoutGenericVarSignatureVertexNode(factory, type);
 
                 // TODO Internal.TypeSystem.TypeFlags.FunctionPointer (Runtime parsing also not yet implemented)
+                // Pretend for now it's an IntPtr, may need to be revisited depending on https://github.com/dotnet/runtime/issues/11354
                 case Internal.TypeSystem.TypeFlags.FunctionPointer:
-                    throw new NotImplementedException("FunctionPointer signature");
+                    return new NativeLayoutEETypeSignatureVertexNode(factory, factory.TypeSystemContext.GetWellKnownType(WellKnownType.IntPtr));
```

Comment:
These are interned, so it would be better to add handling for this case in NodeFactory.NativeLayout.cs (where NewTypeSignatureVertexNode is called).